src/ripple/core/sql_backend/SQLInterface_sqlite.cpp

```diff
@@ -0,0 +1,1280 @@
+
+    virtual bool
+    getHashesByIndex(
+        beast::Journal const& j,
+        LedgerIndex ledgerIndex,
+        uint256& ledgerHash,
+        uint256& parentHash) override;
+
+    virtual std::map<LedgerIndex, std::pair<uint256, uint256>>
+    getHashesByIndex(
+        beast::Journal const& j,
+        LedgerIndex minSeq,
+        LedgerIndex maxSeq) override;
+
+    virtual Json::Value
+    loadTxHistory(Application& app, LedgerIndex startIndex) override;
+
+    virtual NetworkOPs::AccountTxs
+    getAccountTxs(
+        Application& app,
+        LedgerMaster& ledgerMaster,
+        beast::Journal& j,
+        AccountID const& account,
+        std::int32_t minLedger,
+        std::int32_t maxLedger,
+        bool descending,
+        std::uint32_t offset,
+        int limit,
+        bool bUnlimited) override;
+
+    virtual std::vector<NetworkOPs::txnMetaLedgerType>
+    getAccountTxsB(
+        Application& app,
+        beast::Journal& j,
+        AccountID const& account,
+        std::int32_t minLedger,
+        std::int32_t maxLedger,
+        bool descending,
+        std::uint32_t offset,
+        int limit,
+        bool bUnlimited) override;
+
+    virtual void
+    accountTxPage(
+        AccountIDCache const& idCache,
+        std::function<void(std::uint32_t)> const& onUnsavedLedger,
+        std::function<
+            void(std::uint32_t, std::string const&, Blob&&, Blob&&)> const&
+            onTransaction,
+        AccountID const& account,
+        std::int32_t minLedger,
+        std::int32_t maxLedger,
+        bool forward,
+        std::optional<NetworkOPs::AccountTxMarker>& marker,
+        int limit,
+        bool bAdmin,
+        std::uint32_t page_length) override;
+
+    virtual boost::variant<Transaction::pointer, bool>
+    loadTransaction(
+        Application& app,
+        uint256 const& id,
+        boost::optional<ClosedInterval<uint32_t>> const& range,
+        error_code_i& ec) override;
+
+    virtual bool
+    checkDBSpace(DatabaseType dbtype, Config const& config, beast::Journal& j)
+        override;
+
+    virtual int
+    getKBUsedAll(DatabaseType dbtype) override;
+
+    virtual int
+    getKBUsedDB(DatabaseType dbtype) override;
+
+private:
+    Application& app_;
+    std::map<LedgerIndex, DatabaseCon*> txInd2db_;
+    std::map<LedgerIndex, DatabaseCon*> lgrInd2db_;
+    std::unique_ptr<DatabaseCon> lgrdb_, txdb_;
+    std::mutex maps_mutex_;
+
+    LedgerIndex
+    seqToShardIndex(LedgerIndex ledgerSeq)
+    {
+        return app_.getShardStore()->seqToShardIndex(ledgerSeq);
+    }
+
+    LedgerIndex
+    firstLedgerSeq(LedgerIndex shardIndex)
+    {
+        return app_.getShardStore()->firstLedgerSeq(shardIndex);
+    }
+
+    LedgerIndex
+    lastLedgerSeq(LedgerIndex shardIndex)
+    {
+        return app_.getShardStore()->lastLedgerSeq(shardIndex);
+    }
+
+    bool
+    exists(SQLInterface::DatabaseType dtype)
+    {
+        if (dtype == SQLInterface::LEDGER)
```

Comment:
Consider using a switch statement here (but also ok as-is).

Reply:
Done.